draft-jholland-quic-multicast.md

```diff
@@ -262,76 +299,44 @@ MC_CHANNEL_PROPERTIES Frame {
- * Key Length: present when Has Key is set in the Content Field.  Provides the length of the Key field.  It MUST match a valid key length for the AEAD Algorithm.
- * Key: present when Has Key is set in the Content Field, with length given by Key Length.  Used to protect the packet contents of 1-RTT packets for the channel as described in {{RFC9001}}.
-
-   To maintain forward secrecy and prevent malicious clients from decrypting packets long after they have left or were removed from the unicast connection, servers SHOULD periodically send key updates using only unicast.
- * Integrity Hash Algorithm: present when Has Hash Algorithm is set in the Content Field.  The hash algorithm used in integrity frames.
-   * **Author's Note:** Several candidate iana registries, not sure which one to use?  Some have only text for some possibly useful values.  For now we use the first of these:
-     - <https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg>
-     - <https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-18>
-     - (text-only): <https://www.iana.org/assignments/hash-function-text-names/hash-function-text-names.xhtml>
- * Max Rate: Present when Has Max Rate is set in the Content Field.  The maximum rate in Kibps of the payload data for this channel.Channel data MUST NOT exceed this rate over any 5s window, if it does clients SHOULD leave the channel with reason Max Rate Exceeded.
- * Max Idle Time: Present when Has Max Idle Time is set in the Content Field.  The maximum expected idle time of the channel.  If this amount of time passes in a joined channel without data received, clients SHOULD leave the channel with reason Max Idle Time Exceeded.
- * Max Streams: Present when Has Max Streams is set in the Content Field.  The maximum stream ID that might appear in the channel.  If a client joined to this channel can raise its Max Streams limit up to or above this value it SHOULD do so, otherwise it SHOULD leave or decline join for the channel with Max Streams Exceeded.
- * ACK Bundle Size: Present when Has Ack Bundle Size is set in the Content Field.  The minimum number of ACKs a client should send in a single QUIC packet. If the max_ack_delay would force a client to send a packet that only consists of MC_CHANNEL_ACK frames, it SHOULD instead wait with sending until at least the specified number of acknowledgements have been collected. However, the Client MUST send any pending acknowledgements at least once per Max Idle Time to prevent the Server from perceiving the channel as interrupted.
+The properties of a channel MAY change during its lifetime. As such, a server SHOULD wait with sending an MC_CHANNEL_PROPERTIES frame intended to provide the initial properties to a client that hasn't joined the channel yet until immediately before sending the MC_CHANNEL_JOIN frame.
```

Comment:
suggest replacing from "SHOULD ..." with: "SHOULD NOT send properties for channels except those the client has joined or will be imminently asked to join."